### ISSUE TITLE
startup: Failure to revoke tickets that are voted.

### ISSUE THREAD

**Issue by JoeGruffins** [decred/dcrwallet#2052]
I have been using a hot-cold wallet setup to participate in staking for some time now. Just recently, with v1.6.2-3 I am seeing the tickets voted while my cold wallet was offline come up in an error on startup.

```
[ERR] WLLT: Failed to sign revocation for ticket hash abc....def: txscript.SignTxOutput: transaction script fails to execute:: wallet locked: default
[ERR] WLLT: Failed to sign revocation for ticket hash def...abc: txscript.SignTxOutput: transaction script fails to execute:: wallet locked: default
```

Both are tickets that voted int the last couple of weeks. Unlocking the wallet and passing `revoketickets` returns no errors and has no effect. Rescans have no effect. Running on master also has no effect.

**Comment by chappjc** [comment]
Seeing the same lately.

**Comment by davecgh** [comment]
I see this on restarts as well.

**Comment by davecgh** [comment]
For a bit more diagnosis, the culprit seems to be that both voting wallet and ticket buying wallets do not see the vote (or perhaps think it's invalid for some reason when it isn't) and thus try to revoke.

In the case of the voting wallet, since it has the private keys, it creates and attempts to publish a revocation, but that fails because the ticket is already spent by the vote:

> [ERR] WLLT: Failed to send one or more revocations: dcrd.PublishTransactions: rejected transaction <revocation tx hash>: orphan transaction <revocation tx hash> references outputs of unknown or fully-spent transaction <ticket hash>

In the case of the ticket buyer wallet, it does not have the private keys, but it still attempts to publish a revocation and naturally fails with the following indicating it doesn't have the private keys:

> [ERR] WLLT: Failed to sign revocation for ticket hash <ticket hash>: txscript.SignTxOutput: transaction script fails to execute:: invalid operation: account does not record private keys

**Comment by matheusd** [comment]
Not sure whether related or not, but while testing some other dcrwallet PR in simnet I got the following log on a voting only wallet:

```
2021-06-03 17:30:05.130 [INF] SYNC: Connected block 0c240691460418b93cbfeda72aaf873334cc288c1093c9b830f3cf571dc54d5d, height 244, 0 wallet transaction(s)
2021-06-03 17:30:05.159 [INF] WLLT: Voting on block 0c240691460418b93cbfeda72aaf873334cc288c1093c9b830f3cf571dc54d5d (height 244) using ticket a43c2e69f464615c1d85a7d8d91f721ce33db249f11802cd6808d888b9f5d465 (vote hash: 9101403e4fc3a2bae429af450420bbe517a4a507128b1244a7db7d588821f993 bits: 1)
2021-06-03 17:30:05.161 [INF] WLLT: Inserting unconfirmed transaction 9101403e4fc3a2bae429af450420bbe517a4a507128b1244a7db7d588821f993
2021-06-03 17:30:06.189 [INF] SYNC: Connected block 59c7ef1145dc536452391ff9fecddc9cdfcfda776ef4a941b01821d73ba9cbae, height 245, 1 wallet transaction(s)
2021-06-03 17:30:06.189 [INF] WLLT: Revoked ticket a43c2e69f464615c1d85a7d8d91f721ce33db249f11802cd6808d888b9f5d465 with revocation 6b15c78507ac21df51dbcfaa19865956686e3466585146ba0d4f78b851e54238
2021-06-03 17:30:06.190 [ERR] WLLT: Failed to send one or more revocations: dcrd.PublishTransactions: rejected transaction 6b15c78507ac21df51dbcfaa19865956686e3466585146ba0d4f78b851e54238: orphan transaction 6b15c78507ac21df51dbcfaa19865956686e3466585146ba0d4f78b851e54238 references outputs of unknown or fully-spent transaction a43c2e69f464615c1d85a7d8d91f721ce33db249f11802cd6808d888b9f5d465
2021-06-03 17:30:06.190 [INF] WLLT: Inserting unconfirmed transaction 6b15c78507ac21df51dbcfaa19865956686e3466585146ba0d4f78b851e54238
2021-06-03 17:30:07.257 [INF] SYNC: Connected block 7cbc764ceddc47fd99da977851f9f684e710a89a84843442955ba3a022be41ca, height 246, 0 wallet transaction(s)
```

The wallet tried to revoke the ticket _immediately_ after voting on it (down to the same millisecond).

**Comment by davecgh** [comment]
Yep, that is the exact behavior.

**Comment by JoeGruffins** [comment]
I guess I'll look into it further unless someone else is.

**Comment by davecgh** [comment]
A user in support reported they didn't see this until they enabled `settreasurypolicy`.  I haven't tested to see if I can reproduce the issue without it, but I figured I'd post it here in case it is related.

**Comment by JoeGruffins** [comment]
Am able to reproduce on simnet thanks to that hint. It seems to be directly related to whether the voting wallet voted on a treasury spend.

**Comment by JoeGruffins** [comment]
@davecgh It looks like dcrd is falsely reporting these tickets as missed, which is causing them to be treated so in dcrwallet.
https://explorer.dcrdata.org/tx/88329d0f7173559912cc90d4e7815c9516d6004a51ecdfbc773cd3a959510a24
```
dcrctl existsmissedtickets '["f427464a8b0976b2993b6d017eb2168abec23a70a449563dcaf40bf15c80e42c"]'
01
```

**Comment by JoeGruffins** [comment]
linking https://github.com/decred/dcrd/issues/2668

**Comment by davecgh** [comment]
Resolved by decred/dcrd#2780.  It hasn't been merged yet, but I'm going to close this because there is nothing to be done wallet side.
